### PR TITLE
[Components with more than 1 service] LPS-202191 antivirys-async-store - Splitting responsibility of AntivirusAsyncStatisticsManager

### PR DESCRIPTION
**Pull request by rcsdnj**
AntivirusAsyncStatisticsManager was an AntivirusAsyncEventListener and also a JMX component which allowed to view antivirus statistics through JMX visualizatian tools. With this change it now serves only JMX and there's a separate class for listening events and also for updating/providing some antivirus statistics

### CODE REVIEW

File: modules/dxp/apps/antivirus/antivirus-async-store/src/main/java/com/liferay/antivirus/async/store/internal/event/AntivirusStatisticsUpdaterAsyncEventListener.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import com.liferay.portal.kernel.messaging.Message;
+
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * @author Roberto Cassio Silva do Nascimento Junior
+ */
+@Component(service = AntivirusAsyncEventListener.class)
+public class AntivirusStatisticsUpdaterAsyncEventListener
+	implements AntivirusAsyncEventListener {
```
[the remark this answers is not in the document]
Comment (rcsdnj) [reply modules/dxp/apps/antivirus/antivirus-async-store/src/main/java/com/liferay/antivirus/async/store/internal/event/AntivirusStatisticsUpdaterAsyncEventListener.java]: Thanks, Prax. Done!

**Comment by catarineaguiar** [comment]
ci:test:sf

**Comment by catarineaguiar** [comment]
ci:test:relevant

**Comment by Kvale1733** [comment]
No regressions found.

**Review by rafaprax — APPROVED** [review]

**Comment by rafaprax** [comment]
ci:test:sf

**Comment by rafaprax** [comment]
ci:test:relevant

**Comment by Kvale1733** [comment]
ci:test:document-antivirus

**Comment by Kvale1733** [comment]
@rcsdnj Regression found in several document library integration tests. Looks like relevant wasn't catching this so it might not be the latest push that causes this. I found it in the document-antivirus suite

Example)
com.liferay.document.library.util.test.DLDirectoryNameAndFileNameTest.testUpdateFileEntry
```
[beanshell]      [exec] com.liferay.document.library.util.test.DLDirectoryNameAndFileNameTest > testUpdateFileEntry FAILED
[beanshell]      [exec]     java.lang.Exception: Unexpected exception, expected<com.liferay.document.library.kernel.exception.FileNameException> but was<java.lang.NullPointerException>
[beanshell]      [exec]     	at com.liferay.arquillian.extension.junit.bridge.server.TestExecutorRunnable._processExpectedException(TestExecutorRunnable.java:407)
[beanshell]      [exec]     	at com.liferay.arquillian.extension.junit.bridge.server.TestExecutorRunnable.access$400(TestExecutorRunnable.java:57)
[beanshell]      [exec]     	at com.liferay.arquillian.extension.junit.bridge.server.TestExecutorRunnable$3.evaluate(TestExecutorRunnable.java:371)
[beanshell]      [exec]     	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
[beanshell]      [exec]     	at com.liferay.portal.kernel.test.rule.AbstractTestRule$2.e
```

```
[beanshell]      [exec]     Caused by: java.lang.NullPointerException
[beanshell]      [exec]     	at com.liferay.document.library.kernel.antivirus.AntivirusScannerUtil.isActive(AntivirusScannerUtil.java:21)
[beanshell]      [exec]     	at com.liferay.portlet.documentlibrary.store.DLStoreImpl._addFile(DLStoreImpl.java:318)
[beanshell]      [exec]     	at com.liferay.portlet.documentlibrary.store.DLStoreImpl.addFile(DLStoreImpl.java:82)
[beanshell]      [exec]     	at com.liferay.document.li
```
[Log](https://testray.liferay.com/reports/production/logs/2023-12/test-1-27/test-portal-acceptance-pullrequest(master)/10976/modules-integration-mysql57-jdk8/0/5/jenkins-console.txt.gz?authuser=0)

**Comment by Kvale1733** [comment]
ci:close

**Comment by rcsdnj** [comment]
@Kvale1733 , on my local tests, 6 of 7 tests from the test suite were failing before my changes. It persisted the same way, so I think it is not related. 

cc: @rafaprax

**Comment by rcsdnj** [comment]
@Kvale1733 , I also tested the same suite on [this PR I've created for experimentation](https://github.com/rcsdnj/liferay-portal/pull/3). Curiously, it had even more failures than my tests. And the same test you've put as an example [seems to be failing also](https://test-1-40.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/489437/testReport/), so I'll reopen for reevaluation, ok?

**Comment by rcsdnj** [comment]
ci:reopen

**Comment by zhangyan0701** [comment]
ci:test:document-antivirus

**Comment by zhangyan0701** [comment]
Two many tests failed in the document-antivirus test suite, will re-run it just to be safe.

**Comment by zhangyan0701** [comment]
ci:test:document-antivirus

**Comment by dantewang** [comment]
Let's see whether https://github.com/liferay-core-infra/liferay-portal/pull/6387 works.

**Pull request closed without merging**